esp-wifi/src/common_adapter/mod.rs

```diff
@@ -311,6 +311,7 @@ mod log {
 // #define ESP_EVENT_DEFINE_BASE(id) esp_event_base_t id = #id
 static mut EVT: i8 = 0;
 #[no_mangle]
+#[allow(unused_unsafe)]
```

Comment:
I think we should listen to the lints here, it's not unsafe to create a pointer, only to dereference one.

Reply:
Without the `unsafe` it fails with: 
```
error[E0133]: use of mutable static is unsafe and requires unsafe function or block
   --> src/common_adapter/mod.rs:314:52
    |
314 | static mut WIFI_EVENT: esp_event_base_t = addr_of!(EVT);
    |                                                    ^^^ use of mutable static
    |
    = note: mutable statics can be mutated by multiple threads: aliasing violations or data races will cause undefined behavior
```

See https://github.com/esp-rs/esp-hal/actions/runs/10366874177/job/28697094152

Reply:
Interesting, if it is really used why do we need `#[allow(unused_unsafe)]`

Reply:
I think this is relevant for our use case: https://github.com/rust-lang/rust/issues/129578#issuecomment-2309372169 and the recommendation is to add the `allow` if I read it correctly.

BTW this also should (IMO) be a mut pointer which it somehow isn't in esp-wifi-sys.

Reply:
Ah nice find, bit annoying but if that's the recommendation then so be it.